### ISSUE TITLE
Search Improvements

### ISSUE THREAD

**Issue by standardorcawolf** [standardnotes/snjs#270]
Currently Standard Notes has a simple searching protocol and there are minor improvements that would be:

1) all lowercase text would search case insensitive and diacritics. 

That is wolf would match Wolf wOlf, WOLF, WOLF and wölf 

2) An upper case letter anywhere would search directly for that word or sequence.

3) search for 'wolf' would return notes that only contain wolf 

4) search for ''wolf'' (double single quotes) would search for 'wolf' (that means that quotes are nestable)

In other words:

All lower case we do a fuzzy search, when we add an upper case letter we stop searching as fuzzy.

I would be more happy to implement this feature if there is interest for this.

**Comment by mobitar** [comment]
Not sure I'm following. Current functionality is case-insensitive search:

<img width="584" alt="Screen Shot 2021-04-30 at 11 07 32 AM" src="https://user-images.githubusercontent.com/3277844/116722678-545c5500-a9a4-11eb-92f6-acee6994c8b5.png">

**Comment by standardorcawolf** [comment]
# Search enhancement
Exactly case insensitive search works well when we're not exactly sure of what we're searching for.

Since we're not exactly sure what we're searching for we should extend that case insensitivity to accent characters as well.

That is searching `test` should return any notes that contains `tést` or any other accented characters.

However when people search using an upper case letter, they're looking for that exact match. In other words case and accent insensitivity should only be in effect when searching is all lowercase.

That is searching for `Test` should only return pages with `Test` and NOT `Tést` or `test`.

When uppercase we should search for the word or words as is without using any transformation.

Please ignore 3 and 4. That can be a further refinement on search once things such as AND and NOT are added in. I was getting WAY to far ahead of myself.

**Comment by JaspalSuri** [comment]
I actually think it would be better to something along the lines of [3 and 4](https://github.com/standardnotes/snjs/issues/270#issue-872664750). 🙂 Do other apps use [1 and 2](https://github.com/standardnotes/snjs/issues/270#issue-872664750)?

**Comment by standardorcawolf** [comment]
I can't easily find examples of other apps using diacritic insensitive search, however it easily meshes in and is an extension of 1. The idea is from 'The Humane Interface' by Jef Raskin.

However regardless of 2, 1 which is case insensitivity/sensitivity
could be easily implemented and I'd be willing to do that.😁️
